src/auto/file_enumerator.rs

```diff
@@ -0,0 +1,316 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+#[cfg(feature = "futures")]
+use futures::future;
+use gio_sys;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_sys;
+use gobject_sys;
+#[cfg(feature = "futures")]
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::ptr;
+use Cancellable;
+use Error;
+use File;
+use FileInfo;
+
+glib_wrapper! {
+    pub struct FileEnumerator(Object<gio_sys::GFileEnumerator, gio_sys::GFileEnumeratorClass, FileEnumeratorClass>);
+
+    match fn {
+        get_type => || gio_sys::g_file_enumerator_get_type(),
+    }
+}
+
+pub const NONE_FILE_ENUMERATOR: Option<&FileEnumerator> = None;
+
+pub trait FileEnumeratorExt: 'static {
+    fn close<P: IsA<Cancellable>>(&self, cancellable: Option<&P>) -> Result<(), Error>;
+
+    fn close_async<P: IsA<Cancellable>, Q: FnOnce(Result<(), Error>) + Send + 'static>(
+        &self,
+        io_priority: glib::Priority,
+        cancellable: Option<&P>,
+        callback: Q,
+    );
+
+    #[cfg(feature = "futures")]
+    fn close_async_future(
+        &self,
+        io_priority: glib::Priority,
+    ) -> Box_<dyn future::Future<Output = Result<(), Error>> + std::marker::Unpin>;
+
+    fn get_child(&self, info: &FileInfo) -> Option<File>;
+
+    fn get_container(&self) -> Option<File>;
+
+    fn has_pending(&self) -> bool;
+
+    fn is_closed(&self) -> bool;
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn iterate<P: IsA<Cancellable>>(
```

Comment:
This binding is wrong and needs to be handled manually. See the docs how it's supposed to be used from C. I guess we ideally wrap this in a proper Rust iterator.

Or maybe not bind it at all, it's C convenience API and `next_file()` works better for us (and should get an `Iterator` wrapped around it).

Reply:
Yeah I guess ignoring this function and just keeping `next_file()` and its variants would be best. But before merging, please add an

```rust
impl Iterator for FileEnumerator {
    type Item = Result<FileInfo, Error>;

    fn next(&mut self) -> Option<Result<FileInfo, Error>> {
        match self.next_file() {
            Err(err) => Some(Err(err)),
            Ok(file_info) => file_info.map(Ok),
        }
    }
}
```

With the `fallible-iterator` crate this is also good to work with.

Reply:
Added it.